### PR TITLE
Fix(auth): use crypto/subtle to compare strings

### PR DESCRIPTION
**Pull request by till**
Related: #37

### CODE REVIEW

**Comment by friedrichg** [comment]
why did the CI didn't run for this?

**Review by friedrichg — APPROVED** [review]

File: gateway/middleware.go [hunk 1]
```diff
@@ -53,14 +54,21 @@ func (tenant *Tenant) basicAuth(w http.ResponseWriter, r *http.Request) bool {
...
-			return true
-		} else {
-			return false
-		}
+	if !tenant.saveCompare(username, password) {
+		return false
 	}
 
+	r.Header.Set("X-Scope-OrgID", tenant.ID)
+
```
Comment (friedrichg) [inline gateway/middleware.go:62]:
```suggestion
```
Let's remove this blank

**Comment by friedrichg** [comment]
@till thanks! LGTM

**Comment by till** [comment]
> why did the CI didn't run for this?

Maybe required approval? First time contributor.

**Pull request merged**